src/app/public/modules/country-field/country-field.component.ts

```diff
@@ -0,0 +1,295 @@
+   * @default us
+   */
+  @Input()
+  public set defaultCountry(value: string) {
+    if (value !== this._defaultCountry) {
+      this._defaultCountry = value.toLowerCase();
+
+      this.defaultCountryData = this.countries
+        .find(country => country.iso2 === this._defaultCountry);
+
+      this.sortCountriesWithSelectedAndDefault(this.selectedCountry);
+    }
+  }
+
+  public get defaultCountry(): string {
+    return this._defaultCountry;
+  }
+
+  /**
+   * Indicates whether to disable the country field.
+   */
+  @Input()
+  public disabled: boolean = false;
+
+  /**
+   * Emits when the selected country changes.
+   */
+  @Output()
+  public selectedCountryChange: EventEmitter<SkyCountryFieldCountry> = new EventEmitter<SkyCountryFieldCountry>();
+
+  public countries: SkyCountryFieldCountry[];
+
+  public countrySearchForm: FormGroup;
+
+  public isInPhoneField: boolean = false;
+
+  public isInputFocused: boolean = false;
+
+  @ViewChild(SkyAutocompleteInputDirective)
+  public countrySearchAutocompleteDirective: SkyAutocompleteInputDirective;
+
+  public set selectedCountry(newCountry: SkyCountryFieldCountry) {
+    if (this._selectedCountry !== newCountry) {
+      this._selectedCountry = newCountry;
+
+      this.sortCountriesWithSelectedAndDefault(newCountry);
+
+      this.countrySearchForm.get('countrySearch').setValue(this.selectedCountry);
+
+      if (!this.isFirstChange) {
+        this.onChange(newCountry);
+        this.onTouched();
+
+        this.selectedCountryChange.emit(newCountry);
+      }
+
+      // Do not mark the field as "dirty"
+      // if the field has been initialized with a value.
+      if (this.isFirstChange && this.control) {
+        this.control.markAsPristine();
+      }
+
+      if (this.isFirstChange && newCountry !== null) {
+        this.isFirstChange = false;
+      }
+    }
+  }
+
+  public get selectedCountry(): SkyCountryFieldCountry {
+    return this._selectedCountry;
+  }
+
+  private control: AbstractControl;
+
+  private defaultCountryData: SkyCountryFieldCountry;
+
+  private idle: Subject<any> = new Subject();
+
+  private isFirstChange: boolean = true;
+
+  private _defaultCountry: string;
+
+  private _selectedCountry: SkyCountryFieldCountry;
+
+  constructor(
+    private changeDetector: ChangeDetectorRef,
+    private elRef: ElementRef,
+    private formBuilder: FormBuilder,
+    private windowRef: SkyAppWindowRef
+  ) {
+    /**
+     * The json functions here ensures that we get a copy of the array and not the global original.
+     * This ensures that multiple instances of the component don't overwrite the original data.
+     *
+     * We must type the window object as any here as the intl-tel-input library adds its object
+     * to the main window object.
+     */
+    this.countries = JSON.parse(JSON.stringify((window as any)
+      .intlTelInputGlobals.getCountryData()));
+    this.defaultCountryData = this.countries.find(country => country.iso2 === 'us');
```

Comment:
Why are you setting `defaultCountryData` to `us` in the constructor, and then again in the `ngOnInit()`? Are both of these necessary?

Reply:
Nope. Removed the one in the constructor